i3pystatus: 3.33 -> 3.34

igsha opened the pull request:
###### Things done:
- [ ] Tested via `nix.useChroot`.
- [x] Built on platform(s): x86_64 linux.
- [x] Tested compilation of all pkgs that depend on this change.
- [x] Tested execution of binary products.
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).
###### Extra

``` bash
[23:02] igor@igor-pc nixpkgs (i3pystatus) $ nix-shell --pure -I nixpkgs=/home/igor/nixpkgs/ -p python3Packages.i3pystatus

[nix-shell:~/nixpkgs]$ i3pystatus
{"version": 1, "click_events": true}
[
[{"urgent": false, "name": "i3pystatus.clock.Clock", "instance": "139924714783296", "full_text": "Wed 24 Feb 23:04:15", "markup": "none"}]
,[{"urgent": false, "name": "i3pystatus.clock.Clock", "instance": "139924714783296", "full_text": "Wed 24 Feb 23:04:16", "markup": "none"}]
,[{"urgent": false, "name": "i3pystatus.clock.Clock", "instance": "139924714783296", "full_text": "Wed 24 Feb 23:04:17", "markup": "none"}]
,[{"urgent": false, "name": "i3pystatus.clock.Clock", "instance": "139924714783296", "full_text": "Wed 24 Feb 23:04:18", "markup": "none"}]
^C
[nix-shell:~/nixpkgs]$
```

igsha committed:
i3pystatus: 3.33 -> 3.34

FRidh commented on pkgs/top-level/python-packages.nix:
```diff
@@ -5973,16 +5973,15 @@ in modules // {
 
-  i3pystatus = buildPythonPackage rec {
+  i3pystatus = buildPythonApplication rec {
     name = "${pname}-${version}";
```
Good catch. Being a collection of scripts `buildPythonApplication` fits better than `buildPythonPackage`.

That actually also means that this expression needs to be moved elsewhere, somewhere in `pkgs/applications`, because `python-packages.nix` is meant only for packages, not applications.

igsha commented:
@FRidh, moved to `i3` folder in `pkgs/applications`. Also added `extraLibs`. This allows to use extra python modules in i3pystatus configuration file.

FRidh commented on pkgs/applications/window-managers/i3/pystatus.nix:
```diff
@@ -0,0 +1,41 @@
+    makeWrapper ${python3Packages.python.interpreter} $out/bin/${pname}-python-interpreter \
+      --prefix PYTHONPATH : "$PYTHONPATH" \
+      ${ldWrapperSuffix}
+  '';
```
Can `wrapPythonPrograms` that's in `pythonPackages.wrapPython` be used for this @domenkozar  ?
See https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/python-modules/generic/wrap.sh

domenkozar replied:
I don't understand what this line is trying to achieve. Why is it creating an interpreter with these libs?

igsha replied:
@domenkozar, configuration file for the i3pystatus is just a python script that can be run within `i3bar` like this:
```
#i3bar
bar {
    status_command    python ~/.path/to/your/i3pystatus/configuration/file.py
    position          top
    workspace_buttons yes
}
```
In my own i3-configuration I use it in this way:
```
bar {
    status_command i3pystatus-python-interpreter ~/.i3/status-config.py
    position top
    workspace_buttons yes
}
```
where `i3pystatus-python-interpreter` is generated by `postInstall` phase with all necessary libs.
I could use `i3pystatus` application for my purposes, but it is an example application that produces clock time and nothing else. It can't load configuration file like `xmobar` or `i3status`.
I didn't find better solution.

igsha commented:
Can somebody tell me the current status of this merge request?
